commandapi-platforms/commandapi-velocity/commandapi-velocity-plugin/src/main/java/dev/jorel/commandapi/CommandAPIMain.java

```diff
@@ -35,24 +35,17 @@ public CommandAPIMain(ProxyServer server, Logger logger, @DataDirectory Path dat
 		// Try to find the config file
 		Path configFile = dataDirectory.resolve("config.yml");
 
-		// If the config doesn't exist, load it from the resources
-		if(!Files.exists(configFile)) {
-			try {
-				Files.createDirectories(configFile.getParent());
-			} catch (IOException ignored) {
-			}
-
-			try (InputStream defaultConfig = getClass().getClassLoader().getResourceAsStream("config.yml")) {
-				Files.copy(defaultConfig, configFile);
-			} catch (IOException e) {
-				throw new RuntimeException(e);
-			}
-		}
+		// Create or update config
+		VelocityConfigurationAdapter.createDummyInstance().saveDefaultConfig(configFile.getParent().toFile(), configFile.toFile(), logger);
 
 		// Load the file as a yaml node
 		ConfigurationNode configYAML;
 		try {
-			configYAML = YamlConfigurationLoader.builder().path(configFile).build().load();
+			YamlConfigurationLoader loader = YamlConfigurationLoader.builder()
+				.nodeStyle(NodeStyle.BLOCK)
+				.path(configFile)
+				.build();
```

Comment:
Do we need to create the `YamlConfigurationLoader` again here? `VelocityConfigurationAdapter#saveDefaultConfig` creates an identical loader, which is accessible as a field of that record.

Reply:
> Do we need to create the `YamlConfigurationLoader` again here?

Yes. `saveDefaultConfig()` doesn't return anything so the loader used in that method is not accessible. Of course, we could change the return type to the `ConfigurationAdapter` itself, however I do not think it's worth it here because we would have to do the same on Bukkit (as the `saveDefaultConfig` method is inherited from `ConfigurationAdapter`) where it would be absolutely useless.

Reply:
I suppose `CommandAPIMain` could create the `YamlConfigurationLoader` and pass it into `VelocityConfigurationAdapter#createDummyInstance`.